src/repo.rs

```diff
@@ -989,6 +1001,36 @@ impl Repository {
         Ok(())
     }
 
+    /// Merges the given commit(s) into HEAD, writing the results into the working directory.
+    /// Any changes are staged for commit and any conflicts are written to the index. Callers
+    /// should inspect the repository's index after this completes, resolve any conflicts and
+    /// prepare a commit.
+    pub fn merge(&self,
+                 annotated_commits: &[&AnnotatedCommit],
+                 merge_opts: &MergeOptions,
+                 checkout_opts: Option<&mut CheckoutBuilder>) -> Result<(), Error> {
+        unsafe {
+            let mut raw_checkout_opts = mem::zeroed();
+            try_call!(raw::git_checkout_init_options(&mut raw_checkout_opts,
+                                raw::GIT_CHECKOUT_OPTIONS_VERSION));
+            match checkout_opts {
+                Some(c) => c.configure(&mut raw_checkout_opts),
+                None => {}
+            }
+
+            let commit_ptrs: Vec<*const raw::git_annotated_commit> =  annotated_commits.iter().map(|c| {
+                c.raw() as *const raw::git_annotated_commit
+            }).collect();
```

Comment:
Stylistically I'd recommend putting the type annotation at the end and doing something like `.collect::<Vec<_>>()`